pkg/doc/verifiable/credential_jwt.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package verifiable
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/square/go-jose/v3"
+	"github.com/square/go-jose/v3/jwt"
+)
+
+// CredentialJWT is JWT Claims extension by Verifiable Credential.
+type CredentialJWT struct {
+	*jwt.Claims
+
+	Credential *rawCredential `json:"vc,omitempty"`
+}
+
+// Signed creates signed JWT (JWS)
+func (vcc *CredentialJWT) Signed(signatureAlg JWTAlgorithm, privateKey interface{}, keyID string) (string, error) {
+	key := jose.SigningKey{Algorithm: signatureAlg.Jose(), Key: privateKey}
+
+	var signerOpts = &jose.SignerOptions{}
+	signerOpts.WithType("JWT")
+	signerOpts.WithHeader("kid", keyID)
+
+	rsaSigner, err := jose.NewSigner(key, signerOpts)
+	if err != nil {
+		return "", fmt.Errorf("failed to create signer: %w", err)
+	}
+
+	// create an instance of Builder that uses the rsa signer
+	builder := jwt.Signed(rsaSigner).Claims(vcc)
+
+	// validate all ok, sign with the RSA key, and return a compact JWT
+	rawJWT, err := builder.CompactSerialize()
+	if err != nil {
+		return "", fmt.Errorf("failed to sign JWT: %w", err)
+	}
+
+	return rawJWT, nil
+}
+
+// Plain builds unsecured JWT.
+// Currently go-jose does not have the plaintext builder. A workaround solution used here is to build JWS
+// and then clean the signature part of the serialized JWT. That's why a private key is needed.
+// Also, typ and alg are set to none as VC spec required.
+// TODO use plain (unsecured) JWT builder when it will be implemented (https://github.com/square/go-jose/issues/264)
+func (vcc *CredentialJWT) Plain(privateKey interface{}) (string, error) {
```

Comment:
Why is there a privateKey argument? (Even if you are doing a workaround, we shouldn’t have arguments that wouldn’t make sense with the real solution).

Reply:
Fixed in a new version where this workaround is not used anymore.

Reply:
Ok - I assume we are waiting for this new version? @kdimak

Reply:
Yes, I've just pushed it.